backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt

```diff
@@ -29,74 +29,76 @@ private val logger = createLogger("ExceptionHandler")
 /**
  * Registers exception handling.
  */
+@Suppress("LongMethod") // it's fine as this is part of KTor
 fun Application.installExceptionHandling() {
     install(StatusPages) {
-        // TODO define correct logging level policy
-
         exception<InsufficientRightsException> {
-            logger.warn {
+            logger.warn(it) {
                 call.principal<UserPrincipal>()
-                    ?.let { "$it tried to access resource \"${call.request.path()}\" that is not allowed." }
+                    ?.let { user -> "${user.userId} tried to access resource \"${call.request.path()}\" that is not allowed." }
                     ?: "User without principals tried to access the resource ${call.request.path()}."
             }
             call.respond(HttpStatusCode.Forbidden)
         }
 
         exception<CaptchaFailedException> {
-            logger.debug { "Captcha verification failed." }
+            logger.warn(it) { "Captcha verification failed - $it.¬" }
             call.errorResponse(HttpStatusCode.UnprocessableEntity, "Captcha verification failed.")
         }
 
         exception<AuthorizationException> {
+            logger.warn(it) { "Authorization failed - $it." }
             call.respond(HttpStatusCode.Unauthorized)
         }
 
-        exception<EntityNotFoundException> { cause ->
-            logger.debug { cause.message }
-            call.errorResponse(HttpStatusCode.NotFound, cause.message)
+        exception<EntityNotFoundException> {
+            logger.warn(it) { "Entity was not found - $it." }
+            call.errorResponse(HttpStatusCode.NotFound, it.message)
         }
 
-        exception<InvalidSlotCreationRequest> { cause ->
-            logger.debug { "${cause.message} --> ${cause.entity}" }
-            call.errorResponse(HttpStatusCode.BadRequest, cause.message)
+        exception<InvalidSlotCreationRequest> {
+            logger.error(it) { "Invalid slot creation request - $it." }
+            call.errorResponse(HttpStatusCode.BadRequest, it.message)
         }
 
-        exception<NoVaccinationSlotsFoundException> { cause ->
-            call.errorResponse(HttpStatusCode.NotFound, cause.message)
+        exception<NoVaccinationSlotsFoundException> {
+            logger.error(it) { "No vaccination slots found - $it." }
+            call.errorResponse(HttpStatusCode.NotFound, it.message)
         }
 
         jsonExceptions()
 
-        exception<IsinValidationException> { cause ->
-            logger.debug { cause.message }
-            call.errorResponse(HttpStatusCode.NotAcceptable, "Not acceptable: ${cause.message}.")
+        exception<IsinValidationException> {
+            logger.info(it) { "ISIN validation failed - $it." }
```

Comment:
proc jenom info?

Reply:
hele tot otazka, nevim jak casto se bude stavat, ze nam nekdo zada cloveka, co neexistuje, ale asi muzu hodit warn, snad to nebude tolik plasit